### ISSUE TITLE
Support for specifying a file as the body of an upload

### ISSUE THREAD

**Issue by joeapearson** [taverntesting/tavern#410]
I'm aware it's possible to use the `files` property to specify a map of files to be uploaded as part of a multipart file upload.

What I need is the ability to specify the content of a file as the _entire_ body, as described here:

https://docs.microsoft.com/en-us/rest/api/storageservices/put-blob

Here's the sample request that MS provides in their docs:

```
Request Syntax:  
PUT https://myaccount.blob.core.windows.net/mycontainer/myblockblob HTTP/1.1  
  
Request Headers:  
x-ms-version: 2015-02-21  
x-ms-date: <date>  
Content-Type: text/plain; charset=UTF-8  
x-ms-blob-content-disposition: attachment; filename="fname.ext"  
x-ms-blob-type: BlockBlob  
x-ms-meta-m1: v1  
x-ms-meta-m2: v2  
Authorization: SharedKey myaccount:YhuFJjN4fAR8/AmBrqBz7MG2uFinQ4rkh4dscbj598g=  
Content-Length: 11  
  
Request Body:  
hello world
```

(in this case, the content of the file would be "hello world")

I may be missing something, in which case apologies!  So far in my reading it seems that this is not yet supported.   I think tavern would need to [stream the upload](https://2.python-requests.org//en/master/user/advanced/#streaming-uploads) instead of using [multi-part encoded file uploads](https://2.python-requests.org//en/master/user/advanced/#post-multiple-multipart-encoded-files)

**Comment by michaelboulton** [comment]
It's not possible at the moment - I think it might be doable by adding a new key like `file_body` which is exclusive with `files`/`data` and just streaming it like in that example in the requests documentation.

**Comment by michaelboulton** [comment]
Should have been closed with #493
